sdk/ai/ai-projects/src/agents/vectorStores.ts

```diff
@@ -80,3 +81,47 @@ export async function deleteVectorStore(
   }
   return result.body; 
 }
+
+
+/**
+ * Creates a vector store and poll.
+ */
+export async function createVectorStoreAndPoll(
+  context: Client,
+  options?: CreateVectorStoreParameters,
+  sleepIntervalInMs?: number,
+  timeoutInMs: number = 10000,
+): Promise<VectorStoreOutput> {
+  async function updateCreateVectorStorePoll(
+    currentResult?: VectorStoreOutput
+  ): Promise<{ result: VectorStoreOutput; completed: boolean }> {
+    let vectorStore: VectorStoreOutput;
+    if (!currentResult) {
+      vectorStore = await createVectorStore(context, options);
+    } else {
+      vectorStore = await getVectorStore(context, currentResult.id);
+    }
+    return {
+      result: vectorStore,
+      completed: vectorStore.status !== "in_progress",
```

Comment:
vectorStore.status !== "in_progress", if its timeout would the completed return as true ?

Reply:
This is the check used in the python code. I'm not sure if "timed_out" is a status but it would result in true if that is one.